lib/Sema/CSFix.cpp

```diff
@@ -1884,6 +1884,34 @@ std::string TreatEphemeralAsNonEphemeral::getName() const {
   return name;
 }
 
+bool AllowSendingMismatch::diagnose(const Solution &solution,
+                                    bool asNote) const {
+  switch (kind) {
+  case Kind::Parameter: {
+    SendingOnFunctionParameterMismatchFail failure(
+        solution, getFromType(), getToType(), getLocator(), fixBehavior);
+    return failure.diagnose(asNote);
+  }
+  case Kind::Result: {
```

Comment:
It doesn't look like this split brings much to the table, the difference could be handled internally to the diagnostic.